dtfs-central-api/src/v1/controllers/utilisation-report-service/post-remove-fees-from-payment.controller/helpers.ts

```diff
@@ -0,0 +1,44 @@
+import { FeeRecordEntity, UtilisationReportEntity } from '@ukef/dtfs2-common';
+import { TfmSessionUser } from '../../../../types/tfm/tfm-session-user';
+import { executeWithSqlTransaction } from '../../../../helpers';
+import { UtilisationReportStateMachine } from '../../../../services/state-machines/utilisation-report/utilisation-report.state-machine';
+import { InvalidPayloadError } from '../../../../errors';
+
+export const validateAtLeastOneFeeRecordSelected = (selectedFeeRecordIds: number[]) => {
+  if (selectedFeeRecordIds.length === 0) {
+    throw new InvalidPayloadError('No fee records selected.');
+  }
+};
+
+export const validateNotAllFeeRecordsSelected = (selectedFeeRecordIds: number[], totalSelectableFeeRecords: number) => {
+  if (selectedFeeRecordIds.length === totalSelectableFeeRecords) {
+    throw new InvalidPayloadError('Not all fee records can be selected.');
+  }
+};
+
+export const removeFeesFromPayment = async (
+  utilisationReport: UtilisationReportEntity,
+  allFeeRecords: FeeRecordEntity[],
+  selectedFeeRecordIds: number[],
+  user: TfmSessionUser,
+) => {
```

Comment:
I think there is a possibility for confusion around this method being named `removeFeesFromPayment` but then it doesn't actually pass in a payment or anything seemingly related to a payment. I think for this reason I would do some renaming along the lines of:
- `removeFeesFromPayment` -> `removeFeesFromPaymentGroup` (I think at this level it is useful to be clear what is actually happening, i.e. removing the fee record from all payments in the group not just the one)
- `allFeeRecords` -> `feeRecordsInPaymentGroup`
- `selectedFeeRecordIds` -> `idsOfFeeRecordsToRemove`

Reply:
Renamed `removeFeesFromPayment` references in the central API to include `group`, references in the TFM API + UI remain the same (no group reference).

Changes for the last 2 name changes in commit d6b8d3c55e07905a3ac53ccb516fd9eac32ee583